### PR TITLE
Submitting the UT for BaseListFragment class

### PR DESCRIPTION
**Pull request by TKRana**
image

Coverage Gain - 158

### CODE REVIEW

**Comment by carloscaldas-aurea** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java
Comment (carloscaldas-aurea) [inline ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java]: ❗️ This file contains multiple top-level classes

This allows multiple definitions for a class. Which definition gets used is affected by the order in which the source files are passed to the compiler. Having only one top-level class per file guarantees that the class files generated by compilation, and the behavior of the resulting program, are independent of the order in which the source files are passed to the compiler. So Consider the proposed solution.

## The possible solution this case

Consider splitting the current file in two files:

- BaseListFragmentTest.java - containing the class `BaseListFragmentTest`
- TestBaseListFragment.java - containin the class `TestBaseListFragment` 

<a id="c10a6790-50d9-11e8-89e2-ed64d7ac7be5" href="none" codereview-uuid-reserved-tag />

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java
Comment (carloscaldas-aurea) [inline ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java]: ✅ It is nice to see the usage of @Override taking advantage of compiler checking to make sure that the code is actually overriding a method.

**No change suggested, looks good!**
 

<a id="e0397f70-50d9-11e8-89e2-ed64d7ac7be5" href="none" codereview-uuid-reserved-tag />

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java
Comment (carloscaldas-aurea) [inline ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java]: ⚠️  It could be a good idea to import the `static` assertions. This way the code becomes smaller and avoids repetition.  

**[More details](https://confluence.devfactory.com/x/MxNrEw)**
          
## The possible solution in this case

```JAVA
import static org.junit.Assert.assertTrue;
...
assertTrue(isMethodSuccessfullyRun); 
```
 

<a id="06fcbf00-50da-11e8-89e2-ed64d7ac7be5" href="none" codereview-uuid-reserved-tag />

**Review by carloscaldas-aurea — COMMENTED** [review]
Product-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members) completed. 

<table>
  <tr>
    <td>:no_entry_sign:<b>Severes:</b>  0</td>
    <td>:exclamation:<b>Reds:</b>  1</td>
    <td>:warning:️<b>Yellows:</b>  1</td>
    <td>:white_check_mark:<b>Greens:</b>  1</td>
  </tr>
</table>

<details><summary>More information</summary>
  <table>
    <tr>
      <td>:no_entry_sign:</td>
      <td>Severe Findings point out serious violations which can affect stability, performance, maintainability and therefore should be fixed.
        <ul>
          <li>Please note that the merge might have been blocked until all Severe Findings are fixed.</li>
          <li>While it is highly recommended to implement all findings only Severe Findings might have been blocking the merge</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>:exclamation:</td>
      <td>Red Findings represent potential bugs and it is highly recommended to implement them.<br/>
      Not required for the approval of the PR
      </td>
    </tr>
    <tr>
      <td>:warning:️</td>
      <td>Yellow Findings are suggestions to improve the code, warnings and tips<br/>
      Not required for the approval of the PR
      </td>
    </tr>
    <tr>
      <td>:white_check_mark:</td>
      <td>Green Findings highlight good ideas and solutions</td>
    </tr>
    <tr>
      <td>:repeat:</td>
      <td>Repeated instances of the same findings are counted just once and flagged with the repeat icon</td>
    </tr>
  </table>

  Feel free to reply to this comment, (make sure you `@tag` us) so we can start a conversation. The goal here is to learn and grow together and make our products better.

  > NOTE: This review **does not** replace your team's review._
</details>
<br/>
<b>How likely would you recommend the Code Review service to another developer?</b>

Not at all likely \[ [1](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/277&entry.1163746017=1) \] \[ [2](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/277&entry.1163746017=2) \] \[ [3](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/277&entry.1163746017=3) \] \[ [4](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/277&entry.1163746017=4) \] \[ [5](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/277&entry.1163746017=5) \] \[ [6](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/277&entry.1163746017=6) \] \[ [7](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/277&entry.1163746017=7) \] \[ [8](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/277&entry.1163746017=8) \] \[ [9](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/277&entry.1163746017=9) \] \[ [10](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/277&entry.1163746017=10) \] Very likely

> TIP: Press `Ctrl` (or `⌘` in a Mac) to open the survey in a new tab.

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java
Comment (bartosz-rybusinski-aurea) [inline ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java]: Please don't use generic matchers - any(), anyXXXX, etc
we know exactly what values we expect here
in case marcher is combined with verify we can use argument captor @Captor
Apply in all occurrences for all PRs

**Review by bartosz-rybusinski — CHANGES_REQUESTED** [review]

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java
Comment (TKRana) [reply ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java]: done

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java
Comment (TKRana) [reply ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java]: done

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java
Comment (TKRana) [reply ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java]: done

**Comment by TKRana** [comment]
I have implemented all the suggestions. Please continue the review and let me know if any chnges are required.

**Comment by TKRana** [comment]
Can you please complete the review of this PR as it waiting at this state for a long time. Since we are towards the end of the project this PR needs to be completed ASAP.

**Review by bartosz-rybusinski — APPROVED** [review]

**Pull request merged**
